### PR TITLE
Update upstream dependencies to v1.31.0

### PR DESCRIPTION
**Pull request by shizuchanw**
**What this PR does / why we need it**:
Update upstream dependencies to v1.31.0 in `go.mod`
Ran `go mod tidy` to update `go.sum`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # N/A

### CODE REVIEW

**Comment by bryantbiggs** [comment]
we'll need to update the test dependencies (e2e and integration) as well with this change

**Comment by shizuchanw** [comment]
> we'll need to update the test dependencies (e2e and integration) as well with this change

[legacy-cloud-providers](https://github.com/kubernetes/legacy-cloud-providers) is a dependency for the integration tests. When I try to update this dependency I get `go: [k8s.io/legacy-cloud-providers@v0.0.0](http://k8s.io/legacy-cloud-providers@v0.0.0): reading [k8s.io/legacy-cloud-providers/go.mod](http://k8s.io/legacy-cloud-providers/go.mod) at revision v0.31.0: unknown revision v0.31.0`
Seems like the new releases are always done by [k8s-publishing-bot](https://github.com/kubernetes/legacy-cloud-providers/commits?author=k8s-publishing-bot) in a day after the upstream release, but it stopped the automation since v0.31.0-alpha.0; should we manually cut a release for v0.31.0 or leave it at v0.30.0?

**Comment by bryantbiggs** [comment]
lets see what the tests say, but I suspect they'll fail

/ok-to-test

**Comment by bryantbiggs** [comment]
your `go.sum` seems to be off - `go mod tidy` should resolve this

File: go.mod [hunk 1]
```diff
@@ -8,30 +8,30 @@ require (
 	github.com/gofrs/flock v0.8.1
 	github.com/google/go-cmp v0.6.0
 	github.com/manifoldco/promptui v0.9.0
-	github.com/prometheus/client_golang v1.16.0
+	github.com/prometheus/client_golang v1.19.1
```
Comment (bryantbiggs) [inline go.mod:11]: if we don't require these other packages to be updated, might be better to save for a separate PR and just update the `k8s.io/*` deps

**Comment by shizuchanw** [comment]
> your `go.sum` seems to be off - `go mod tidy` should resolve this

I met this issue locally as well, solved it by deleting `go.sum` entirely and ran `go mod tidy` to create a new `go.sum`, but seems like that's not the solution as it won't pass CI. Looking into how to solve this... Some related post I found: https://stackoverflow.com/questions/66007082/go-sum-are-different-when-using-go-mod-vendor-download

**Comment by bryantbiggs** [comment]
using your patch, removing the `go.sum` files and re-running `go mod tidy`, this is what I end up with on my machine

```diff
diff --git a/go.sum b/go.sum
index 7382c2ef..2a3bcb3a 100644
--- a/go.sum
+++ b/go.sum
@@ -17,7 +17,7 @@ github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSs
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc h1:U9qPSI2PIWSS1VwoXQT9A3Wy9MM3WgvqSxFWenqJduM=
 github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
-github.com/emicklei/go-restful/v3 v3.11.1 h1:PZ8lyY5omynnq9Y2BHKmNqkC3AnPMES2e2bg7slcfzw=
+github.com/emicklei/go-restful/v3 v3.11.1 h1:S+9bSbua1z3FgCnV0KKOSSZ3mDthb5NyEPL5gEpCvyk=
 github.com/emicklei/go-restful/v3 v3.11.1/go.mod h1:6n3XBCmQQb25CM2LCACGz8ukIrRry+4bhvbpWn3mrbc=
 github.com/frankban/quicktest v1.14.6 h1:7Xjx+VpznH+oBnejlPUj8oUpdxnVs4f8XU8WnHkI4W8=
 github.com/frankban/quicktest v1.14.6/go.mod h1:4ptaffx2x8+WTWXmUCuVU6aPUX1/Mz7zb5vbUoiM6w0=
diff --git a/tests/integration/go.sum b/tests/integration/go.sum
index 4162e260..f4a756a0 100644
--- a/tests/integration/go.sum
+++ b/tests/integration/go.sum
@@ -37,7 +37,7 @@ github.com/distribution/reference v0.5.0 h1:/FUIFXtfc/x2gpa5/VGfiGLuOIdYa1t65IKK
 github.com/distribution/reference v0.5.0/go.mod h1:BbU0aIcezP1/5jX/8MP0YiH4SdvB5Y4f/wlDRiLyi3E=
 github.com/dustin/go-humanize v1.0.1 h1:GzkhY7T5VNhEkwH0PVJgjz+fX1rhBrR7pRT3mDkpeCY=
 github.com/dustin/go-humanize v1.0.1/go.mod h1:Mu1zIs6XwVuF/gI1OepvI0qD18qycQx+mFykh5fBlto=
-github.com/emicklei/go-restful/v3 v3.11.1 h1:PZ8lyY5omynnq9Y2BHKmNqkC3AnPMES2e2bg7slcfzw=
+github.com/emicklei/go-restful/v3 v3.11.1 h1:S+9bSbua1z3FgCnV0KKOSSZ3mDthb5NyEPL5gEpCvyk=
 github.com/emicklei/go-restful/v3 v3.11.1/go.mod h1:6n3XBCmQQb25CM2LCACGz8ukIrRry+4bhvbpWn3mrbc=
 github.com/felixge/httpsnoop v1.0.4 h1:NFTV2Zj1bL4mc9sqWACXbQFVBBg2W3GPvqp8/ESS2Wg=
 github.com/felixge/httpsnoop v1.0.4/go.mod h1:m8KPJKqk1gH5J9DgRY2ASl2lWCfGKXixSwevea8zH2U=
```

**Comment by shizuchanw** [comment]
> using your patch, removing the `go.sum` files and re-running `go mod tidy`, this is what I end up with on my machine
> 
> ```diff
> diff --git a/go.sum b/go.sum
> index 7382c2ef..2a3bcb3a 100644
> --- a/go.sum
> +++ b/go.sum
> @@ -17,7 +17,7 @@ github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSs
>  github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
>  github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc h1:U9qPSI2PIWSS1VwoXQT9A3Wy9MM3WgvqSxFWenqJduM=
>  github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
> -github.com/emicklei/go-restful/v3 v3.11.1 h1:PZ8lyY5omynnq9Y2BHKmNqkC3AnPMES2e2bg7slcfzw=
> +github.com/emicklei/go-restful/v3 v3.11.1 h1:S+9bSbua1z3FgCnV0KKOSSZ3mDthb5NyEPL5gEpCvyk=
>  github.com/emicklei/go-restful/v3 v3.11.1/go.mod h1:6n3XBCmQQb25CM2LCACGz8ukIrRry+4bhvbpWn3mrbc=
>  github.com/frankban/quicktest v1.14.6 h1:7Xjx+VpznH+oBnejlPUj8oUpdxnVs4f8XU8WnHkI4W8=
>  github.com/frankban/quicktest v1.14.6/go.mod h1:4ptaffx2x8+WTWXmUCuVU6aPUX1/Mz7zb5vbUoiM6w0=
> diff --git a/tests/integration/go.sum b/tests/integration/go.sum
> index 4162e260..f4a756a0 100644
> --- a/tests/integration/go.sum
> +++ b/tests/integration/go.sum
> @@ -37,7 +37,7 @@ github.com/distribution/reference v0.5.0 h1:/FUIFXtfc/x2gpa5/VGfiGLuOIdYa1t65IKK
>  github.com/distribution/reference v0.5.0/go.mod h1:BbU0aIcezP1/5jX/8MP0YiH4SdvB5Y4f/wlDRiLyi3E=
>  github.com/dustin/go-humanize v1.0.1 h1:GzkhY7T5VNhEkwH0PVJgjz+fX1rhBrR7pRT3mDkpeCY=
>  github.com/dustin/go-humanize v1.0.1/go.mod h1:Mu1zIs6XwVuF/gI1OepvI0qD18qycQx+mFykh5fBlto=
> -github.com/emicklei/go-restful/v3 v3.11.1 h1:PZ8lyY5omynnq9Y2BHKmNqkC3AnPMES2e2bg7slcfzw=
> +github.com/emicklei/go-restful/v3 v3.11.1 h1:S+9bSbua1z3FgCnV0KKOSSZ3mDthb5NyEPL5gEpCvyk=
>  github.com/emicklei/go-restful/v3 v3.11.1/go.mod h1:6n3XBCmQQb25CM2LCACGz8ukIrRry+4bhvbpWn3mrbc=
>  github.com/felixge/httpsnoop v1.0.4 h1:NFTV2Zj1bL4mc9sqWACXbQFVBBg2W3GPvqp8/ESS2Wg=
>  github.com/felixge/httpsnoop v1.0.4/go.mod h1:m8KPJKqk1gH5J9DgRY2ASl2lWCfGKXixSwevea8zH2U=
> ```


This is the original error I got before I removed `go.sum` and re-ran `go mod tidy`:
```
go: downloading github.com/emicklei/go-restful/v3 v3.11.1
verifying github.com/emicklei/go-restful/v3@v3.11.1: checksum mismatch
        downloaded: h1:PZ8lyY5omynnq9Y2BHKmNqkC3AnPMES2e2bg7slcfzw=
        go.sum:     h1:S+9bSbua1z3FgCnV0KKOSSZ3mDthb5NyEPL5gEpCvyk=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.
```
Seems like my local environment is having a different download hash than yours and the original hash... trying on cloud desktop now

**Comment by shizuchanw** [comment]
@bryantbiggs Made a revision that keeps the original hash and seems like all the tests are passing, could you take another look? Thank you!!!

**Comment by bryantbiggs** [comment]
/lgtm 

cc @micahhausler / @jyotimahapatra
